skorch/tests/callbacks/test_early_stopping.py

```diff
@@ -0,0 +1,170 @@
+"""Tests for early stopping callback"""
+
+import numpy as np
+import pytest
+
+
+class TestEarlyStopping:
+
+    @pytest.fixture
+    def early_stopping_cls(self):
+        from skorch.callbacks import EarlyStopping
+        return EarlyStopping
+
+    @pytest.fixture
+    def epoch_scoring_cls(self):
+        from skorch.callbacks import EpochScoring
+        return EpochScoring
+
+    @pytest.fixture
+    def net_clf_cls(self):
+        from skorch import NeuralNetClassifier
+        return NeuralNetClassifier
+
+    @pytest.fixture
+    def broken_classifier_module(self, classifier_module):
+        """Return a classifier that does not improve over time."""
+        class BrokenClassifier(classifier_module):
+            def forward(self, x):
+                return super().forward(x) * 0 + 0.5
+        return BrokenClassifier
+
+    def test_typical_use_case_nonstop(
+            self, net_clf_cls, classifier_module, classifier_data,
+            early_stopping_cls):
+        patience = 5
+        max_epochs = 8
+        early_stopping_cb = early_stopping_cls(patience=patience)
+
+        net = net_clf_cls(
+            classifier_module,
+            callbacks=[
+                early_stopping_cb,
+            ],
+            max_epochs=max_epochs,
+        )
+        net.fit(*classifier_data)
+
+        ok_run = (
+            len(net.history) == max_epochs
+        )
+        assert ok_run
+
+    def test_typical_use_case_stopping(
+            self, net_clf_cls, broken_classifier_module, classifier_data,
+            early_stopping_cls):
+        patience = 5
+        max_epochs = 8
+        early_stopping_cb = early_stopping_cls(patience=patience)
+
+        net = net_clf_cls(
+            broken_classifier_module,
+            callbacks=[
+                early_stopping_cb,
+            ],
+            max_epochs=max_epochs,
+        )
+        net.fit(*classifier_data)
+
+        bad_run = (
```

Comment:
Can we not be more precise here? I.e. instead of `len(net.history) != max_epochs`, we know that `assert len(net.history) == patience + 1 < max_epochs`.